frost-core/src/frost/identifier.rs

```diff
@@ -14,183 +14,91 @@ use crate::{Ciphersuite, Error, Field, Group, Scalar};
 /// over, corresponding to some x-coordinate for a polynomial f(x) = y.  MUST NOT be zero in the
 /// field, as f(0) = the shared secret.
 #[derive(Copy, Clone)]
-pub struct Identifier<C: Ciphersuite>(pub(crate) Scalar<C>);
+pub struct Identifier<C>(u16, PhantomData<C>);
 
-impl<C> AsRef<Scalar<C>> for Identifier<C>
+impl<C> Identifier<C>
 where
     C: Ciphersuite,
 {
-    fn as_ref(&self) -> &Scalar<C> {
-        &self.0
+    // Convert the identifier to a Scalar.
+    //
+    // Ideally this would be a From<Identifier<C>> for Scalar<C> impl, but rustc
+    // doesn't like that
+    pub(crate) fn to_scalar(self) -> Scalar<C> {
+        // Classic left-to-right double-and-add algorithm that skips the first bit 1 (since
+        // identifiers are never zero, there is always a bit 1), thus `sum` starts with 1 too.
```

Comment:
Are we sure that we cannot create an instance of `Identifier` with the value of zero? If not sure, should we do a check here? That would require this method to be a `Result<Scalar<C>, Error>` if we do that

Reply:
The only code that can build an `Identifier` is in `identifier.rs`, and there only the `impl<C> TryFrom<u16> for Identifier<C>` builds it. So we should be fine.

However there is no reason to not follow your suggestion since all callers of `to_scalar` already return `Result`s.

So I added the check in b5dd0ce72d